### PR TITLE
Allow expressions in process variables

### PR DESCRIPTION
**Pull request by rbarroso**
Related issue: https://github.com/Activiti/Activiti/issues/3744

### CODE REVIEW

File: activiti-core/activiti-spring-process-extensions/src/main/java/org/activiti/spring/process/variable/types/VariableType.java [hunk 1]
```diff
@@ -16,15 +16,16 @@
...
 import org.activiti.engine.ActivitiException;
 
 /**
- * Base variable type for types as defined in extension json files.
- * Used to validate variables against definition.
+ * Base variable type for types as defined in extension json files. Used to validate variables against definition.
  */
 public abstract class VariableType {
 
+    private static String expressionPattern = "^\\$\\{(.*)[\\}]$";
```
Comment (pmartinezga) [inline activiti-core/activiti-spring-process-extensions/src/main/java/org/activiti/spring/process/variable/types/VariableType.java]:
```suggestion
    private static String expressionPattern = "^\\$\\{(.|\\n)*[\\}]$";
```
If we want to support multiline expressions, then we should change the pattern like this

Comment (mmartinadanx) [reply activiti-core/activiti-spring-process-extensions/src/main/java/org/activiti/spring/process/variable/types/VariableType.java]: Use here a compiled pattern instead
```suggestion
    private static final Pattern EXPRESSION_PATTERN = Pattern.compile("^\\$\\{(.|\\n)*[\\}]$");
```

File: activiti-core/activiti-spring-process-extensions/src/main/java/org/activiti/spring/process/variable/types/VariableType.java [hunk 2]
```diff
@@ -40,4 +41,11 @@ public void setName(String name) {
     public Object parseFromValue(Object value) throws ActivitiException {
         return value;
     }
+
+    protected boolean isExpression(Object var) {
+        if (Objects.nonNull(var) && var.getClass().isAssignableFrom(String.class) && Pattern.compile(expressionPattern).matcher((CharSequence) var).matches()) {
```
Comment (mmartinadanx) [inline activiti-core/activiti-spring-process-extensions/src/main/java/org/activiti/spring/process/variable/types/VariableType.java]: NO if required
```suggestion
        return (Objects.nonNull(var) && var.getClass().isAssignableFrom(String.class) && EXPRESSION_PATTERN.matcher((CharSequence) var).matches());
```

**Review by mmartinadanx — APPROVED** [review]
Nice! just 2 minor comments

File: activiti-core/activiti-spring-process-extensions/src/main/java/org/activiti/spring/process/variable/types/VariableType.java (hunk 1 above)
Comment (rbarroso) [reply activiti-core/activiti-spring-process-extensions/src/main/java/org/activiti/spring/process/variable/types/VariableType.java]: Done

File: activiti-core/activiti-spring-process-extensions/src/main/java/org/activiti/spring/process/variable/types/VariableType.java (hunk 2 above)
Comment (rbarroso) [reply activiti-core/activiti-spring-process-extensions/src/main/java/org/activiti/spring/process/variable/types/VariableType.java]: Thanks. Simplification addressed in last commit

**Pull request merged**